### PR TITLE
Various analysis & QOL improvements

### PR DESCRIPTION
**Pull request by sdobbs**
A few different partially related changes:

* New histogram and cut actions for particles with displaced vertices
* New OPTIONAL ReactionFilter options to help reduce file size
* Reduce the number of informational lines printed out to the screen by eliminating unncessary geometry printouts & only printing other messages once per run.  (plans are being discussed to reduce the XML error messages)

### CODE REVIEW

**Comment by gluex** [comment]
Test status for this pull request: SUCCESS
 
  Summary: [/work/halld/pull_request_test/halld_recon^sdobbs_new_reactionfilter_updates/tests/summary.txt](https://halldweb.jlab.org/pull_request_test/halld_recon^sdobbs_new_reactionfilter_updates/tests-summary.txt)
 Logs: [/work/halld/pull_request_test/halld_recon^sdobbs_new_reactionfilter_updates/tests/log](https://halldweb.jlab.org/pull_request_test/halld_recon^sdobbs_new_reactionfilter_updates/tests-logs)
 
 Build log: [/work/halld/pull_request_test/halld_recon^sdobbs_new_reactionfilter_updates/make_sdobbs_new_reactionfilter_updates.log](https://halldweb.jlab.org/pull_request_test/halld_recon^sdobbs_new_reactionfilter_updates/make_sdobbs_new_reactionfilter_updates.log)
 Build report: [/work/halld/pull_request_test/halld_recon^sdobbs_new_reactionfilter_updates/report_sdobbs_new_reactionfilter_updates.txt](https://halldweb.jlab.org/pull_request_test/halld_recon^sdobbs_new_reactionfilter_updates/report_sdobbs_new_reactionfilter_updates.txt)
 Location of build: /work/halld/pull_request_test/halld_recon^sdobbs_new_reactionfilter_updates

File: src/plugins/Analysis/ReactionFilter/DReaction_factory_ReactionFilter.h [hunk 1]
```diff
@@ -32,6 +32,10 @@ class DReaction_factory_ReactionFilter : public jana::JFactory<DReaction>
 		{
 			// This is so that the created DReaction objects persist throughout the life of the program instead of being cleared each event. 
 			SetFactoryFlag(PERSISTANT);
+
```
Comment (jrstevenjlab) [inline src/plugins/Analysis/ReactionFilter/DReaction_factory_ReactionFilter.h:35]: These options would apply to all reactions in an analysis launch, right?  Would it be useful/practical to set these as command line flags that could be specified for individual reactions to reduce the file size?

Comment (sdobbs) [reply src/plugins/Analysis/ReactionFilter/DReaction_factory_ReactionFilter.h:35]: I was thinking of implementing those options in the future.   The general philosophy has been to apply common loose selections on an analysis launch which don't appreciably reject backgrounds, so I took that as the starting point.

File: src/libraries/HDGEOMETRY/DGeometry.cc [hunk 2]
```diff
@@ -1771,7 +1771,8 @@ bool DGeometry::GetCCALZ(double &z_ccal) const
    bool good = Get("//section/composition/posXYZ[@volume='ComptonEMcal']/@X_Y_Z", ComptonEMcalpos);
 
    if(!good){
-      _DBG_<<"Unable to retrieve ComptonEMcal position."<<endl;
+	  // NEED TO RETHINK ERROR REPORTING FOR OPTIONAL DETECTOR ELEMENTS
```
Comment (jrstevenjlab) [inline src/libraries/HDGEOMETRY/DGeometry.cc:1774]: Is this "rethink" about the cases where a detector is not in the XML geometry and the Get() function reports a warning?  For example 
`src/JANA/JGeometryXML.cc:348 Node or attribute not found for xpath "//section/composition/posXYZ[@volume='ComptonEMcal']/@X_Y_Z".`

It's not for this PR, but should that be a separate issue?

Comment (sdobbs) [reply src/libraries/HDGEOMETRY/DGeometry.cc:1774]: Right, the error messages are coming from the JANA level, so once we decide on a change on that level, we can revisit how to handle these cases.  We can open an issue on this or not - I don't have a strong opinion.

Enjoy the more compact command line output!

**Pull request merged**